packages/debug/src/browser/view/debug-configuration-select.tsx

```diff
@@ -0,0 +1,209 @@
+            this.selectDynamicConfigFromQuickPick(providerType);
+        } else {
+            const [name, type, request, workspaceFolderUri, providerType] = InternalDebugSessionOptions.parseValue(value);
+            this.manager.current =
+                this.manager.find
+                    (
+                        { name, type, request },
+                        workspaceFolderUri,
+                        providerType === 'undefined' ? undefined : providerType
+                    );
+        }
+    };
+
+    protected toPickValue(providerType: string): string {
+        return DebugConfigurationSelect.PICK + providerType;
+    }
+
+    protected parsePickValue(value: string): string {
+        return value.slice(DebugConfigurationSelect.PICK.length);
+    }
+
+    protected async selectDynamicConfigFromQuickPick(providerType: string): Promise<void> {
+        const configurationsOfProviderType = (await this.manager.provideDynamicDebugConfigurations()).find(entry => entry.type === providerType);
+        if (!configurationsOfProviderType) {
+            return;
+        }
+        const { configurations } = configurationsOfProviderType;
+
+        const picks: IDynamicPickItem[] = [];
+        for (const configuration of configurations) {
+            picks.push({
+                label: configuration.name,
+                configurationType: configuration.type,
+                request: configuration.request,
+                providerType
+            });
+        }
+
+        if (picks.length === 0) {
+            return;
+        }
+
+        const quickPick = this.quickInputService.createQuickPick<IDynamicPickItem>();
+        quickPick.items = picks;
+        quickPick.placeholder = nls.localizeByDefault('Select Launch Configuration');
+        quickPick.show();
+
+        const selected: IDynamicPickItem | undefined = await new Promise(resolve => {
+            // If the user presses `Escape` then `quickPick.onDidAccept` will fire
+            // and `quickPick.activeItems` will be empty.
+            quickPick.onDidAccept(() => {
+                resolve(quickPick.activeItems[0]);
+            });
+        });
+
+        quickPick.dispose();
+
+        if (!selected) {
+            return;
+        }
+
+        const selectedConfiguration = { name: selected.label, type: selected.configurationType, request: selected.request };
+        this.manager.current = this.manager.find(selectedConfiguration, undefined, selected.providerType);
+    }
+
+    protected refreshDebugConfigurations = async () => {
+        const configsPerType = await this.manager.provideDynamicDebugConfigurations();
+        const providerTypes = [];
+        for (const { type, configurations } of configsPerType) {
+            if (configurations.length > 0) {
+                providerTypes.push(type);
+            }
+        }
+        this.setState({ providerTypes });
+    };
+
+    protected renderOptions(): React.ReactNode {
+        let index = 0;
```

Comment:
It's usually discourage to use indices as keys in React if it can be avoided. Is there a way to create a unique, persistent identifier from the information used to create the `option`?